test/services/authorization.service.test.js

```diff
@@ -0,0 +1,81 @@
+const {ConfigurationService} =  require("../../services/configurationService");
+const {AuthorizationService} =  require("../../services/authorization-service");
+const {USER} =  require("../../crdc-datahub-database-drivers/constants/user-constants");
+const ERROR = require("../../constants/error-constants");
+
+
+describe('authorization service test', () => {
+
+    let configurationService;
+    let authorizationService;
+    let pbacDefaults;
+    let user;
+
+    beforeAll(() => {
+        configurationService = new ConfigurationService();
+        pbacDefaults = [
+            {
+                role: USER.ROLES.ADMIN,
+                permissions: [
+                    {
+                        _id: "fake",
+                        scopes: ["failed test"]
+                    },
+                    {
+                        _id: "data_submission:submitted",
+                        scopes: ["default"]
+                    }
+                ]
+            }
+        ]
+        configurationService.getPBACByRoles = jest.fn().mockReturnValue(pbacDefaults);
+        authorizationService = new AuthorizationService(configurationService);
+        user = {
+            role: USER.ROLES.ADMIN,
+            permissions: [
+                "data_submission:cancelled:none",
+                'data_submission:submitted:all'
+            ]
+        };
+    });
+
+    test("/test read scope from permission", () => {
+        let permission = "data_submission:submitted"
+        expect(authorizationService.getPermissionScope(user, permission)).toStrictEqual(["all"])
```

Comment:
I am trying to understand. When you insert multiple permissions, why is returning only **all**? There are different permissions.

Reply:
This is testing that it reads from the mock response. The mock response is a stand-in for reading from the database so this test is really just making sure that when no scopes are provided that the output matches the specified mock response for the database call